tests/cli/test_server.py

```diff
@@ -0,0 +1,38 @@
+from click.testing import CliRunner
+
+from strawberry.cli.commands.server import server as cmd_server
+
+import uvicorn
+import hupper
+
+from unittest.mock import Mock
+
+
+def test_cli_cmd_server(monkeypatch):
+
+    def mock_uvicorn_run(*args, **kwargs):
+        exit(0)
+
+    def mock_hupper_start_reloader(*args, **kwargs):
+        return MockReloader()
+
+    # Mock of uvicorn.run
+    monkeypatch.setattr(uvicorn, "run", Mock(wraps=mock_uvicorn_run))
+    # Mock to prevent the reloader from kicking in
+    monkeypatch.setattr(hupper, "start_reloader", Mock(wraps=mock_hupper_start_reloader))
+    runner = CliRunner()
+
+    result = runner.invoke(cmd_server, ['tests.cli.helpers.sample_schema'], "")
+    assert result.exit_code == 0
+
+    # We started the reloader
+    hupper.start_reloader.assert_called_once
+    uvicorn.run.assert_called_once
```

Comment:
Eventually would suggest further mocking to assert uvicorn was called with the right GraphQL object but that's a lot of hoisting and mocking.

Since GraphQL's "contract" with uvicorn is extensively tested elsewhere we can reasonably expect it to be successful.

Likely the server command would need to be refactored to allow more function responses to be hooked in to.

Reply:
uhm, I usually tend to use mocker (we might need to add it to the deps) with pytest and then do something like this:

```
def test_cli_cmd_server(mocker):
   uv_patch = mocker.patch('path.server.uvicorn.run')
   uv_patch.side_effect = ... # or uv_patch.return_value = 
```

would you be ok with trying this approach?

In addition to that I think we need to call `assert_called_once` like this `uvicorn.run.assert_called_once()`

That fixed we can merge this! :)